examples/Input/Input.ino

```diff
@@ -0,0 +1,28 @@
+/*
+ Controlling a servo position using serial input
+ by Hamza Khalid https://github.com/hmzakhalid
+
+*/
+
+#include <Servo.h>
+
+Servo myservo;  // create servo object to control a servo
+
+
+int pos = 0;    // initial Position of the Shaft (max is 180)
+
+void setup() {
+  myservo.attach(9);  // attaches the servo on pin 9 to the servo object
+  Serial.begin(9600);  //opens serial port, sets data rate to 9600 bps
+  Serial.println("Enter Position from 1 to 180"); 
+}
+
+
+void loop() {
+  
+  if(Serial.available() > 0){                       // checks if there is any input on the serial monitor
+    pos = Serial.read();               // Reading the input from the serial monitor and storing it in'pos'
```

Comment:
This doesn't work as advertised. Even when you input numbers, Serial Monitor sends the character representation of those numbers.

Say the user enters the number 42 in the Serial Monitor input field. With the default line ending setting of "Newline", Serial.read() is then going to return three bytes: 52 (the [ASCII decimal code](https://www.asciitable.com/) for the character '4'), 50 (the ASCII decimal code for the character '2'), 10 (the ASCII decimal code for '\n').

Please always thoroughly test your code before submitting a pull request.

Reply:
i used Serial.parseInt() instead of read(). but that Always Returns  0 after a specific timeout period (default 1000ms) . Is there a way i can remove the timeout period?

Reply:
The situation depends on which setting is selected in the Serial Monitor's line ending menu:
- **No line ending**: The 1000 ms timeout must pass before the number sent by the user is returned, making the interface feel sluggish. The solution to this is to use [`Serial.setTimeout()`](https://www.arduino.cc/reference/en/language/functions/communication/serial/settimeout) to set a reasonable timeout. 10 bits/character * 3 characters / 9600 bits/second = .003125 s so you can set the timeout to a duration that is not noticeable to a human.
- **Newline** or **Carriage return** or **Both NL & CR**: The first call to `Serial.parseInt()` returns the number sent by the user immediately, but the next call returns 0 (because the line ending(s) are left in the receive buffer and `Serial.parseInt()` returns 0 when no valid number was found) after the timeout. The solution to avoiding the 0 is to add this code after the call to `Serial.parseInt()`:
    ```c++
    delay(2);  // wait for all the serial data to come in
    // dump line endings from the received data
    while (Serial.available()) {
      Serial.read();
    }
    ```
    That will clear the receive buffer of the line ending character(s).

Reply:
Oh, I didn't know there was a menu for the line ending. I was wondering why does it add a '\n' to every input. 
Alright, i get it now.  `Serial.read()`  reads the new line or carriage return from the stream and it gets dumped. then there's nothing left in the stream.. i guess there's 0? and 0 is a valid number so nothing gets returned.
Thanks So much for helping. I checked the code on my Arduino Uno and its working!